File: solutions/csharp/grains/Grains.cs
public static class Grains
{
    public static ulong Square(int n)
    {
        if (n is < 1 or > 64)
            throw new ArgumentOutOfRangeException(nameof(n));

        return (ulong) Math.Pow(2, n - 1);
    }

    public static ulong Total() =>
        (ulong) Math.Pow(2, 64);
}

File: solutions/fsharp/grains/Grains.fs
module Grains

let square (n: int): Result<uint64,string> =
    match n with
    | _ when n >= 1 && n <= 64 -> 2.0 ** float (n - 1) |> uint64 |> Ok
    | _ -> Error "square must be between 1 and 64"
  
let total: Result<uint64,string> = 2.0 ** 64.0 - 1.0 |> uint64 |> Ok
